superset-frontend/src/visualizations/TimeTable/SortNumberWithMixedTypes.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Row } from 'react-table';
+
+const sortNumberWithMixedTypes = (rowA: Row, rowB: Row, columnId: string) => {
+  const rowAVal = rowA.values[columnId].props['data-value'];
+  const rowBVal = rowB.values[columnId].props['data-value'];
```

Comment:
Can we move `Row` and `columnId` out of this utility function to make it more generic? I'd imagine it can be useful in other places as well.

```ts
import { JSONPrimitive } from '@superset-ui/core';

export default function sortNumberWithMixedTypes(
  valueA: JSONPrimitive | undefined,
  valueB: JSONPrimitive | undefined
) {
}
```

Reply:
`sortType` is prop from `react-table useSortBy`: https://github.com/TanStack/react-table/blob/alpha/docs/api/useSortBy.md 
I am not sure I can change my function signature?

Reply:
You can always have wrapper function for `sortType`.

Reply:
```
ESLint: React Hook "useCallback" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.(react-hooks/rules-of-hooks)
```

Reply:
You can move the wrapper outside of the component as a simple one liner in the component file without using useCallback.

Point is, such utility functions should not depend on too specific input data types if we want to maximize reusability.